### PR TITLE
refactor: Deconstruct bdba grouping

### PR DESCRIPTION
**Pull request by zkdev**
**What this PR does / why we need it**:
Get rid of unused legacy code.
Preparation for streamlined bdba filtering and localBlob support.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

File: protecode/scanning.py [hunk 1]
```diff
@@ -35,23 +34,6 @@
...
-    '''
-    r = resource_group[0]
-    return tuple((
-        r.component.name,
-        (r.resource.name, r.resource.version),
-        r.resource.type,
-    ))
-
-
 def find_related_groups(
```
Comment (8R0WNI3) [inline protecode/scanning.py:37]: `find_related_groups` is also a leftover and can be removed.

File: protecode/scanning.py [hunk 2]
```diff
@@ -190,97 +169,86 @@ def iter_vulnerabilities_with_assessments(
 
     def scan_requests(
```
Comment (8R0WNI3) [inline protecode/scanning.py:170]: As this function does only yield one scan request by now, let us rename it to `scan_request` and don't use a generator anymore.

File: protecode/scanning.py [hunk 3]
```diff
@@ -394,37 +362,32 @@ def raise_on_error(exception):
...
         ))
-        # todo: deduplicate/merge assessments
+
         assessments = self.iter_products(
             products_to_import_from=products_to_import_from,
             use_product_cache=use_product_cache,
             delete_inactive_products_after_seconds=delete_inactive_products_after_seconds,
         )
 
         for scan_request in self.scan_requests(
```
Comment (8R0WNI3) [inline protecode/scanning.py:389]: Also, we can drop looping here.

File: protecode/scanning.py [hunk 4]
```diff
@@ -568,180 +531,17 @@ def result_matches(resource: cm.Resource, result: pm.AnalysisResult):
 def _retrieve_existing_scan_results(
```
Comment (8R0WNI3) [inline protecode/scanning.py:531]: Remove leading underscore since this function is not for private use anymore (-> referenced in bdba cli)

**Review by 8R0WNI3 — CHANGES_REQUESTED** [review]

**Review by 8R0WNI3 — APPROVED** [review]
/lgtm

**Review by ccwienk — APPROVED** [review]

**Pull request merged**